solana/programs/example-native-token-transfers/src/instructions/redeem.rs

```diff
@@ -22,6 +22,10 @@ pub struct Redeem<'info> {
     pub payer: Signer<'info>,
 
     // NOTE: this works when the contract is paused
+    #[account(
+        constraint = config.next_transceiver_id != 0 @ NTTError::NoRegisteredTransceivers,
+        constraint = config.enabled_transceivers.is_empty() @ NTTError::NoRegisteredTransceivers,
```

Comment:
`redeem()` will fail if there aren't enough votes/attestations on a message and it's not possible to have attestations without transceivers, so we might as well fail early and with a more accurate error message.